numpy/_core/src/multiarray/item_selection.c

```diff
@@ -2045,6 +2062,7 @@ PyArray_SearchSorted(PyArrayObject *op1, PyObject *op2,
         Py_DECREF(dtype);
         return NULL;
     }
+    
 
     /* need ap2 as contiguous array and of right dtype (note: steals dtype reference) */
     ap2 = (PyArrayObject *)PyArray_CheckFromAny(op2, dtype,
```

Comment:
Now we do a full conversion to an array already above, which is OK, but it would be nice to just use `op2_tmp` (or name it `ap2` above already and modify it here).
That might need a branch though.  If we got a Python scalar, then we probably need to use this code.